packages/vm/lib/runTx.ts

```diff
@@ -121,6 +122,21 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
   const caller = tx.getSenderAddress()
   debug(`New tx run hash=${opts.tx.hash().toString('hex')} sender=${caller.toString()}`)
 
+  if (block._common.eips().includes(2929)) {
+    // Add origin and precompiles to warm addresses
+    for (let addressInt = 1; addressInt <= numPrecompiles; addressInt++) {
+      // Check if precompile exists on the current Common of the Block
+      if (getPrecompile(new Address(Buffer.from(addressInt)), block._common)) {
+        state.addWarmAddress(new Address(Buffer.from(addressInt)))
```

Comment:
This is not so beautiful to re-create this address space for the precompiles, we already have the `precompiles` const in `precompiles/index.ts` with all the addresses available, and this is also already exported.

Can you please use this here for address creation? Then there is also no need to do this extra `numPrecompiles` export.

Reply:
Should have left a comment here. This is for forward compatibility. Geths implementation only marks current active precompiles to be warm. If we do not implement something like I created here (I agree it is not beautiful currently) then we get a consensus bug, for instance if we call a BLS precompile. Geth would mark it as cold, while we mark it as warm address. (GetPrecompile checks if precompile is available at current HF)

Reply:
Ah ok, thanks. Can you then either:

a) add an additional `getPrecompiles(common: Comon)` function to `precompiles/index.ts` just looping over `precompiles` and return the active ones analog to `getPrecompile()? or

b) minimal solution: add least do the `numPrecompiles` count dynamically by doing a `precompiles.length + 1` in `precompiles/index.ts`, so that we do not introduce a relatively error prone new static constant there

This whole logic really doesn't belong there and we should at some point move to `Common` (not now though likely).

Reply:
I will go for option (a). #727 is related here